crates/hir/src/lib.rs

```diff
@@ -6343,3 +6352,42 @@ where
         self(item)
     }
 }
+
+pub fn resolve_absolute_path(
+    db: &dyn HirDatabase,
+    mut segments: impl Iterator<Item = Symbol>,
+) -> impl Iterator<Item = ItemInNs> {
+    segments
+        .next()
+        .and_then(|crate_name| {
+            let krate = *db.all_crates().iter().find(|&krate| {
```

Comment:
This doesn't handle if we have the same name in here multiple times, e.g. different versions of the same crate

Reply:
That is indeed a problem. I think the best solution will to resolve to all of them.